### PR TITLE
Change process arguments

### PR DESCRIPTION
**Pull request by dalance**
This is my proposal about #7.
The changes are below:

- Add optional output path for #7.
- Change the type of path to `T: AsRef<Path>`.
- Rewrite `build.rs` of examples.

I think that `String` is not suitable to represent a file path because the path may not be valid UTF-8 on some platforms.
If the type is `T: AsRef<Path>`, `process` can get both `Path` and `PathBuf`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,8 +29,7 @@ use std::path::Path;
 
 fn main() {
     let path = Path::new("src").join("target.l");
-    let path = path.to_str().unwrap().to_string();
-    if let Err(e) = rflex::process(path) {
+    if let Err(e) = rflex::process(path, None) {
```
Comment (suma) [inline README.md]: Should we show `OUT_DIR` environment variable in README.md such as examples, or not?

**Review by suma — APPROVED** [review]
Thank you for your PR! Looks nice and I could check the behavior.
I wrote one question.

File: README.md (hunk 1 above)
Comment (dalance) [reply README.md]: I forgot it. I think `OUT_DIR` is required because this is an example of `build.rs`.

Comment (suma) [reply README.md]: :+1:

**Comment by dalance** [comment]
Thanks!

**Pull request merged**